sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3336,6 +3336,35 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession {
       checkAnswer(df5, Array.empty[Row])
     }
   }
+
+  test("SPARK-29505: desc columnname - case insensitive search") {
+    withSQLConf(SQLConf.CASE_SENSITIVE.key -> "false") {
+      withTable("customer") {
+        sql("CREATE TABLE customer(id INT, name STRING, CName STRING, address STRING," +
+          " city STRING, pin INT, country STRING)")
+        sql("INSERT INTO customer VALUES(1,'Alfred','Maria','Obere Str 57'," +
+          "'Berlin',12209,'Germany')")
+        sql("INSERT INTO customer VALUES(2,'Ana','trujilo','Adva de la'," +
+          "'Maxico D.F.',05021,'Maxico')")
+        sql("INSERT INTO customer VALUES(3,'Antonio','Antonio Moreno','Mataderos 2312'," +
+          "'Maxico D.F.',05023,'Maxico')")
```

Comment:
Do we need to populate all data? It seems that we only use `CName` and the other are ignored completely.